### PR TITLE
ASA add redundancy properties

### PR DESCRIPTION
**Pull request by jmcgill298**

### CODE REVIEW

File: pyntc/devices/asa_device.py [hunk 1]
```diff
@@ -25,11 +25,16 @@
...
+RE_SHOW_FAILOVER_GROUPS = re.compile(r"Group\s+\d+\s+State:\s+(.+?)\s*$", re.M)
+RE_SHOW_FAILOVER_STATE = re.compile(r"(?:Primary|Secondary)\s+-\s+(.+?)\s*$", re.M)
+
+
 @fix_docs
 class ASADevice(BaseDevice):
     """Cisco ASA Device Implementation."""
 
     vendor = "cisco"
+    active_redundancy_states = {None, "active"}
```
Comment (mundruid) [inline pyntc/devices/asa_device.py:37]: Why not make this data structure a list?

File: test/unit/test_devices/test_asa_device.py [hunk 2]
```diff
@@ -304,6 +306,75 @@ def test_count_teardown(self):
...
+        (asa_module.CommandError("show failover", r"% invalid command"), None),
+    ),
+    ids=(ACTIVE, "standby", "disabled", "active_active", "active_standby", "standby_active", "none"),
+)
+def test_redundancy_state(side_effect, expected, asa_show):
+    device = asa_show([side_effect])
+    actual = device.redundancy_state
+    assert actual == expected
+
+
```
Comment (mundruid) [inline test/unit/test_devices/test_asa_device.py:377]: The tests look great and I think they cover all possible combinations of states.

**Review by mundruid — APPROVED** [review]
I have one question: why did you use a dictionary for the set of active redundancy states?  Otherwise it looks good!

File: pyntc/devices/asa_device.py (hunk 1 above)
Comment (jvanderaa) [reply pyntc/devices/asa_device.py:37]: I like the usage of the set.

**Review by jvanderaa — APPROVED** [review]
LG

File: pyntc/devices/asa_device.py (hunk 1 above)
Comment (jmcgill298) [reply pyntc/devices/asa_device.py:37]: @mundruid The purpose of the data structure is to do lookups (is x in y?), which performs better in sets than lists

**Pull request merged**